### ISSUE TITLE
Container restarts when 'executing setup script for NORDVPN'

### ISSUE THREAD

**Issue by tonystower** [haugene/docker-transmission-openvpn#2742]
### Is there a pinned issue for this?

- [X] I have read the pinned issues and could not find my issue

### Is there an existing or similar issue/discussion for this?

- [X] I have searched the existing issues
- [X] I have searched the existing discussions

### Is there any comment in the documentation for this?

- [X] I have read the documentation, especially the FAQ and Troubleshooting parts

### Is this related to a provider?

- [X] I have checked the provider repo for issues
- [X] My issue is NOT related to a provider

### Are you using the latest release?

- [X] I am using the latest release

### Have you tried using the dev branch latest?

- [X] I have tried using dev branch

### Docker run config used

{
    "AppArmorProfile": "",
    "Args": [
        "/etc/openvpn/start.sh"
    ],
    "Config": {
        "AttachStderr": false,
        "AttachStdin": false,
        "AttachStdout": false,
        "Cmd": [
            "dumb-init",
            "/etc/openvpn/start.sh"
        ],
        "Domainname": "",
        "Entrypoint": null,
        "Env": [
            "PUID=1000",
            "PGID=100",
            "OPENVPN_PROVIDER=NORDVPN",
            "OPENVPN_USERNAME=XXXXX",
            "OPENVPN_PASSWORD=XXXXX",
            "LOCAL_NETWORK=192.168.0.0/24",
            "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
            "OPENVPN_OPTS=",
            "GLOBAL_APPLY_PERMISSIONS=true",
            "TRANSMISSION_HOME=/config/transmission-home",
            "TRANSMISSION_RPC_PORT=9091",
            "TRANSMISSION_RPC_USERNAME=",
            "TRANSMISSION_RPC_PASSWORD=",
            "TRANSMISSION_DOWNLOAD_DIR=/data/completed",
            "TRANSMISSION_INCOMPLETE_DIR=/data/incomplete",
            "TRANSMISSION_WATCH_DIR=/data/watch",
            "CREATE_TUN_DEVICE=true",
            "ENABLE_UFW=false",
            "UFW_ALLOW_GW_NET=false",
            "UFW_EXTRA_PORTS=",
            "UFW_DISABLE_IPTABLES_REJECT=false",
            "PEER_DNS=true",
            "PEER_DNS_PIN_ROUTES=true",
            "DROP_DEFAULT_ROUTE=",
            "WEBPROXY_ENABLED=false",
            "WEBPROXY_PORT=8118",
            "WEBPROXY_USERNAME=",
            "WEBPROXY_PASSWORD=",
            "LOG_TO_STDOUT=false",
            "HEALTH_CHECK_HOST=google.com",
            "SELFHEAL=false",
            "REVISION=81277fb6f8ae6aeb57f913172d4d45c7af8c0b31",
            "NORDVPN_COUNTRY=gb",
            "NORDVPN_CATEGORY=legacy_p2p"
        ],
        "ExposedPorts": {
            "8118/tcp": {},
            "9091/tcp": {}
        },
        "Healthcheck": {
            "Interval": 60000000000,
            "Test": [
                "CMD-SHELL",
                "/etc/scripts/healthcheck.sh"
            ]
        },
        "Hostname": "b54f08dce4f8",
        "Image": "haugene/transmission-openvpn:dev",
        "Labels": {
            "autoheal": "true",
            "org.opencontainers.image.created": "2023-09-05T11:38:13.353Z",
            "org.opencontainers.image.description": "Docker container running Transmission torrent client with WebUI over an OpenVPN tunnel",
            "org.opencontainers.image.licenses": "GPL-3.0",
            "org.opencontainers.image.ref.name": "ubuntu",
            "org.opencontainers.image.revision": "81277fb6f8ae6aeb57f913172d4d45c7af8c0b31",
            "org.opencontainers.image.source": "https://github.com/haugene/docker-transmission-openvpn",
            "org.opencontainers.image.title": "docker-transmission-openvpn",
            "org.opencontainers.image.url": "https://github.com/haugene/docker-transmission-openvpn",
            "org.opencontainers.image.version": "5.2.0"
        },
        "OnBuild": null,
        "OpenStdin": false,
        "StdinOnce": false,
        "Tty": false,
        "User": "",
        "Volumes": {
            "/config": {},
            "/data": {}
        },
        "WorkingDir": ""
    },
    "Created": "2023-11-11T12:52:49.462498969Z",
    "Driver": "overlay2",
    "ExecIDs": null,
    "GraphDriver": {
        "Data": {
            "LowerDir": "/mnt/dietpi_userdata/docker-data/overlay2/764984bb37148db6624a52a100c975e5e561be3535b229d855ae1c96fefc819a-init/diff:/mnt/dietpi_userdata/docker-data/overlay2/0dd21239b7f57f19da9f950da1e9bce3c22929d3a6840942b3efd8281e053a38/diff:/mnt/dietpi_userdata/docker-data/overlay2/6dcea14e21d9fe08f2818bfb6580b18781f42a73ea2613b81bcef1e3dafc53c2/diff:/mnt/dietpi_userdata/docker-data/overlay2/38a3d40fda2bce6b864c733cb2b6bb2bdf9597e32cd6712b36d179927c270e71/diff:/mnt/dietpi_userdata/docker-data/overlay2/f4de3cd2bd42a4cb87956a63b492bfa16e87d0c26d95116acf537863bb63ee78/diff:/mnt/dietpi_userdata/docker-data/overlay2/3de95f7c980fcc3479e5119c683797b85c2a705ce1f7dd8b1520e90913e045ae/diff:/mnt/dietpi_userdata/docker-data/overlay2/dd918b078cd69473bd15af4397f6f08def6d69d9b96ead2eac2aebe0775d3173/diff:/mnt/dietpi_userdata/docker-data/overlay2/1763570fedb6d44155c6f6985a7bd424c78c2e40a5f5665e33723dc9e8c47c23/diff:/mnt/dietpi_userdata/docker-data/overlay2/5ba695d961904ecc10cb1a435d3cbf90396496eda2c49c054250a8b49558a617/diff:/mnt/dietpi_userdata/docker-data/overlay2/84449cfda2016403652b9d367095b839d2912a5baec4f4d92c183582e4dca6d1/diff:/mnt/dietpi_userdata/docker-data/overlay2/fd072a8f9212b445601bc14a7ad1e79643d4ccf68b3fe750057f3880d8efe2fc/diff:/mnt/dietpi_userdata/docker-data/overlay2/a7cec3fbd6363180685eab032e7136cbd64cc199cee92192e14fb50bef08caf4/diff",
            "MergedDir": "/mnt/dietpi_userdata/docker-data/overlay2/764984bb37148db6624a52a100c975e5e561be3535b229d855ae1c96fefc819a/merged",
            "UpperDir": "/mnt/dietpi_userdata/docker-data/overlay2/764984bb37148db6624a52a100c975e5e561be3535b229d855ae1c96fefc819a/diff",
            "WorkDir": "/mnt/dietpi_userdata/docker-data/overlay2/764984bb37148db6624a52a100c975e5e561be3535b229d855ae1c96fefc819a/work"
        },
        "Name": "overlay2"
    },
    "HostConfig": {
        "AutoRemove": false,
        "Binds": [
            "/mnt/bittorrent/BitTorrent:/data",
            "/mnt/bittorrent/BitTorrent:/config"
        ],
        "BlkioDeviceReadBps": null,
        "BlkioDeviceReadIOps": null,
        "BlkioDeviceWriteBps": null,
        "BlkioDeviceWriteIOps": null,
        "BlkioWeight": 0,
        "BlkioWeightDevice": null,
        "CapAdd": [
            "AUDIT_WRITE",
            "CHOWN",
            "DAC_OVERRIDE",
            "FOWNER",
            "FSETID",
            "KILL",
            "MKNOD",
            "NET_BIND_SERVICE",
            "NET_RAW",
            "SETFCAP",
            "SETGID",
            "SETPCAP",
            "SETUID",
            "SYS_CHROOT"
        ],
        "CapDrop": [
            "AUDIT_CONTROL",
            "BLOCK_SUSPEND",
            "DAC_READ_SEARCH",
            "IPC_LOCK",
            "IPC_OWNER",
            "LEASE",
            "LINUX_IMMUTABLE",
            "MAC_ADMIN",
            "MAC_OVERRIDE",
            "NET_ADMIN",
            "NET_BROADCAST",
            "SYSLOG",
            "SYS_ADMIN",
            "SYS_BOOT",
            "SYS_MODULE",
            "SYS_NICE",
            "SYS_PACCT",
            "SYS_PTRACE",
            "SYS_RAWIO",
            "SYS_RESOURCE",
            "SYS_TIME",
            "SYS_TTY_CONFIG",
            "WAKE_ALARM"
        ],
        "Cgroup": "",
        "CgroupParent": "",
        "CgroupnsMode": "private",
        "ConsoleSize": [
            0,
            0
        ],
        "ContainerIDFile": "",
        "CpuCount": 0,
        "CpuPercent": 0,
        "CpuPeriod": 0,
        "CpuQuota": 0,
        "CpuRealtimePeriod": 0,
        "CpuRealtimeRuntime": 0,
        "CpuShares": 0,
        "CpusetCpus": "",
        "CpusetMems": "",
        "DeviceCgroupRules": null,
        "DeviceRequests": null,
        "Devices": [],
        "Dns": [],
        "DnsOptions": [],
        "DnsSearch": [],
        "ExtraHosts": [],
        "GroupAdd": null,
        "IOMaximumBandwidth": 0,
        "IOMaximumIOps": 0,
        "IpcMode": "private",
        "Isolation": "",
        "Links": null,
        "LogConfig": {
            "Config": {},
            "Type": "journald"
        },
        "MaskedPaths": [
            "/proc/asound",
            "/proc/acpi",
            "/proc/kcore",
            "/proc/keys",
            "/proc/latency_stats",
            "/proc/timer_list",
            "/proc/timer_stats",
            "/proc/sched_debug",
            "/proc/scsi",
            "/sys/firmware"
        ],
        "Memory": 0,
        "MemoryReservation": 0,
        "MemorySwap": 0,
        "MemorySwappiness": null,
        "NanoCpus": 0,
        "NetworkMode": "bridge",
        "OomKillDisable": null,
        "OomScoreAdj": 0,
        "PidMode": "",
        "PidsLimit": null,
        "PortBindings": {
            "9091/tcp": [
                {
                    "HostIp": "",
                    "HostPort": "9091"
                }
            ]
        },
        "Privileged": true,
        "PublishAllPorts": false,
        "ReadonlyPaths": [
            "/proc/bus",
            "/proc/fs",
            "/proc/irq",
            "/proc/sys",
            "/proc/sysrq-trigger"
        ],
        "ReadonlyRootfs": false,
        "RestartPolicy": {
            "MaximumRetryCount": 0,
            "Name": "unless-stopped"
        },
        "Runtime": "runc",
        "SecurityOpt": [
            "label=disable"
        ],
        "ShmSize": 67108864,
        "UTSMode": "",
        "Ulimits": null,
        "UsernsMode": "",
        "VolumeDriver": "",
        "VolumesFrom": null
    },
    "HostnamePath": "/mnt/dietpi_userdata/docker-data/containers/a323a0e439a32d515ace47832c72bfb9fa2569b506abb993f635d6d48184ee2f/hostname",
    "HostsPath": "/mnt/dietpi_userdata/docker-data/containers/a323a0e439a32d515ace47832c72bfb9fa2569b506abb993f635d6d48184ee2f/hosts",
    "Id": "a323a0e439a32d515ace47832c72bfb9fa2569b506abb993f635d6d48184ee2f",
    "Image": "sha256:1f78bec66be1f9c119f83059aaf7bb6dbd95580cf50e6406c9b2d6bfd6f46314",
    "LogPath": "",
    "MountLabel": "",
    "Mounts": [
        {
            "Destination": "/data",
            "Mode": "",
            "Propagation": "rprivate",
            "RW": true,
            "Source": "/mnt/bittorrent/BitTorrent",
            "Type": "bind"
        },
        {
            "Destination": "/config",
            "Mode": "",
            "Propagation": "rprivate",
            "RW": true,
            "Source": "/mnt/bittorrent/BitTorrent",
            "Type": "bind"
        }
    ],
    "Name": "/transmission-openvpn",
    "NetworkSettings": {
        "Bridge": "",
        "EndpointID": "",
        "Gateway": "",
        "GlobalIPv6Address": "",
        "GlobalIPv6PrefixLen": 0,
        "HairpinMode": false,
        "IPAddress": "",
        "IPPrefixLen": 0,
        "IPv6Gateway": "",
        "LinkLocalIPv6Address": "",
        "LinkLocalIPv6PrefixLen": 0,
        "MacAddress": "",
        "Networks": {
            "bridge": {
                "Aliases": null,
                "DriverOpts": null,
                "EndpointID": "",
                "Gateway": "",
                "GlobalIPv6Address": "",
                "GlobalIPv6PrefixLen": 0,
                "IPAMConfig": {},
                "IPAddress": "",
                "IPPrefixLen": 0,
                "IPv6Gateway": "",
                "Links": null,
                "MacAddress": "",
                "NetworkID": "a7d8af0a92c882cc1407b8eaefeac40dad1118bc0a3179e124c3987ae08ac585"
            }
        },
        "Ports": {},
        "SandboxID": "18899a65a58c049262b94722f9d78b034933f99b4bb738daf192bb4802e3b141",
        "SandboxKey": "/var/run/docker/netns/18899a65a58c",
        "SecondaryIPAddresses": null,
        "SecondaryIPv6Addresses": null
    },
    "Path": "dumb-init",
    "Platform": "linux",
    "Portainer": {
        "ResourceControl": {
            "Id": 28,
            "ResourceId": "a323a0e439a32d515ace47832c72bfb9fa2569b506abb993f635d6d48184ee2f",
            "SubResourceIds": [],
            "Type": 1,
            "UserAccesses": [],
            "TeamAccesses": [],
            "Public": false,
            "AdministratorsOnly": true,
            "System": false
        }
    },
    "ProcessLabel": "",
    "ResolvConfPath": "/mnt/dietpi_userdata/docker-data/containers/a323a0e439a32d515ace47832c72bfb9fa2569b506abb993f635d6d48184ee2f/resolv.conf",
    "RestartCount": 202,
    "State": {
        "Dead": false,
        "Error": "",
        "ExitCode": 35,
        "FinishedAt": "2023-11-11T16:11:36.812136526Z",
        "Health": {
            "FailingStreak": 0,
            "Log": [],
            "Status": "unhealthy"
        },
        "OOMKilled": false,
        "Paused": false,
        "Pid": 0,
        "Restarting": true,
        "Running": true,
        "StartedAt": "2023-11-11T16:11:35.14663486Z",
        "Status": "restarting"
    }
}

### Current Behavior

Container has been working flawlessly for months, but for the last few days, I have been unable to access the Transmission web app.
It turns out that the container is failing (and restarting) when executing the setup script for NordVPN.

### Expected Behavior

Container to start normally, connect to NordVPN and allow connection to Transmission

### How have you tried to solve the problem?

1) pulled latest image
2) pulled dev
3) pulled beta
4) creating new container
5) checked NORDVPN username and passwords

### Log output

[_transmission-openvpn_logs-2.txt](https://github.com/haugene/docker-transmission-openvpn/files/13325951/_transmission-openvpn_logs-2.txt)

### HW/SW Environment

```markdown
- RaspberryPi 400
- OS: DietPi 8.2.3
- Docker: 24.0.7
- Portainer 2.19.1
```

**Comment by pkishino** [comment]
Please try older versions of the container.. the logs don’t really make sense here, seems like it just decided to stop instead of running the script for no reason.. can you try with debug logging please?

**Comment by tonystower** [comment]
Thanks. I have tried with 5.0.2 and 4.3.2, both of which exhibited the same behaviour.

I have followed the debug procedure here:  https://haugene.github.io/docker-transmission-openvpn/debug/#try_running_the_container_with_an_invalid_setup

And received this output:
Starting container with revision: 81277fb6f8ae6aeb57f913172d4d45c7af8c0b31
TRANSMISSION_HOME is currently set to: /config/transmission-home
Creating TUN device /dev/net/tun
Using OpenVPN provider: PIA
Running with VPN_CONFIG_SOURCE auto
Provider PIA has a bundled setup script. Defaulting to internal config
Executing setup script for PIA
Downloading OpenVPN config bundle openvpn into temporary file /tmp/tmp.Y2OkeSeUsm
curl: (35) error:0A0003E8:SSL routines::reason(1000)

This suggests an SSL error? Thought I can't understand why...
